db/fixtures/007_web_hook_event_types.rb

```diff
@@ -1,76 +1,197 @@
 # frozen_string_literal: true
 
 WebHookEventType.seed do |b|
-  b.id = WebHookEventType::TOPIC
-  b.name = "topic"
+  b.id = WebHookEventType::TYPES[:topic][:created]
```

Comment:
IMO it's a little awkward to have to type out this kind of lookup all the time (`WebHookEventType::TYPES[:topic][:created]`), too many brackets and colons...will try to think of something else. Not a huge deal if it's just merged as is tho

Reply:
Maybe something like this in the model?


```ruby

  def self.define_type_methods
    TYPES.each do |type, events|
      events.each do |event_name, event_value|
        define_singleton_method("#{type}_#{event_name}") { TYPES[type][event_name] }
      end
    end
  end

  define_type_methods
```

Then you can do this:

```ruby
brc
Loading development environment (Rails 7.0.7)
[1] pry(main)> WebHookEventType.topic_created
=> 101
[2] pry(main)> WebHookEventType.user_promoted_created
=> 1601
[3] pry(main)>
```

Reply:
I am not that big fan of metaprogramming unless benefits are clearly visible. In that case, TYPES constant can be easily found and developer s clearly understand nested hash. Defining methods on the fly is much harder to grep. I read once about the idea of putting a comment about defined methods to make it searchable, like:
```
# def topic_created
# def user_promoted_created
```

Reply:
I hear your concern about this, and I do somewhat agree since this often annoys me too, we need to strike a balance between different devex concerns, and is resilient to typos.

Maybe we can just make the scope part of the type, instead of another level of the hash, so we change to this:

```ruby
# old
  b.id = WebHookEventType::TYPES[:chat][:message_restored]
# new
  b.id = WebHookEventType::TYPES[:chat_message_restored]
```

That way at least it's less awkward to type out than the nested way.